pkg/k8s/watchers/node.go

```diff
@@ -117,3 +135,104 @@ func (k *K8sWatcher) GetK8sNode(_ context.Context, nodeName string) (*v1.Node, e
 	}
 	return nodeInterface.(*v1.Node).DeepCopy(), nil
 }
+
+// ciliumNodeUpdater implements the subscriber.Node interface and is used
+// to keep CiliumNode objects in sync with the node ones.
+type ciliumNodeUpdater struct {
+	k8sWatcher         *K8sWatcher
+	localNode          nodediscovery.LocalNode
+	kvStoreNodeUpdater nodediscovery.KVStoreNodeUpdater
+}
+
+func NewCiliumNodeUpdater(k8sWatcher *K8sWatcher, kvStoreNodeUpdater nodediscovery.KVStoreNodeUpdater, localNode nodediscovery.LocalNode) *ciliumNodeUpdater {
+	return &ciliumNodeUpdater{
+		k8sWatcher:         k8sWatcher,
+		kvStoreNodeUpdater: kvStoreNodeUpdater,
+		localNode:          localNode,
+	}
+}
```

Comment:
Ideally, the interfaces are defined where they are being used (within this file), but given that we've discussed a refactor after this is merged offline anyway, we can address this then.

Reply:
Interfaces moved to `pkg/k8s/watchers/node.go` :+1: